### PR TITLE
EdkRepo: Create unit tests for the module‑level functions in edk_mani…

### PR DESCRIPTION
**Pull request by VictorMlVillarreal**
…fest_validation.py and modularize them if needed

Implemented unit tests and documentation for the module‑level functions in edk_manifest_validation.py to improve clarity, reliability, and maintainability. Some internal methods were refactored to extract logic into smaller helper functions, making the codebase easier to test.

Changes:
-Added unit tests for the the module‑level functions in edk_manifest_validation.py module, covering the main execution paths and edge cases. 
-Created the corresponding documentation file EdkManifestValidation_TestCases.md. 
-Extracted logic from validate_manifestrepo(global_manifest_directory, verify_archived=False) and validate_manifestfiles(manifestfile_list=[]) into some dedicated helper functions called _collect_file_validation_results(manifest_filepath), _resolve_project_manifest_path(ci_index_xml, global_manifest_directory, project), and _collect_project_validation_results(manifest_filepath, project, project_list, ci_index_filename). 
-All new tests rely on mocked dependencies and do not perform filesystem or network operations. 
-Updated internal references to use the refactored helpers.

Fixes: #349

### CODE REVIEW

File: edkrepo_manifest_parser/edk_manifest_validation.py [hunk 1]
```diff
@@ -53,102 +53,92 @@ def list_entries(self, project, project_list):
...
 def validate_manifestrepo(global_manifest_directory, verify_archived=False):
+    """Validate all manifest files referenced by CiIndex.xml; return a dict mapping filepath to result lists."""
     manifestfile_validation = {}
-    # Open CiIndex.xml and parse the data
     ci_index_filename = os.path.join(global_manifest_directory, 'CiIndex.xml')
     ci_index_xml = CiIndexXml(ci_index_filename)
-    # Check every entry in the CiIndex.xml file to verify consistency
     project_list = ci_index_xml.project_list
     if verify_archived:
         project_list.extend(ci_index_xml.archived_project_list)
```
Comment (ashedesimone) [inline edkrepo_manifest_parser/edk_manifest_validation.py]: By extending the list as is we are risking mutating the original source causing unanticipated consequences I would define project list as follows:
`project_list = list(ci_index_xml.project_list)`
